crates/storage/src/schema/revision_0056.rs

```diff
@@ -0,0 +1,127 @@
+use std::time::Instant;
+
+use anyhow::Context;
+use rusqlite::params;
+
+pub(crate) fn migrate(tx: &rusqlite::Transaction<'_>) -> anyhow::Result<()> {
+    tx.execute_batch(
+        r"
+        CREATE TABLE contract_addresses (
+            id INTEGER PRIMARY KEY,
+            contract_address BLOB
+        );
+        CREATE INDEX contract_addresses_contract_address ON contract_addresses (contract_address);
+        CREATE TABLE storage_addresses (
+            id INTEGER PRIMARY KEY,
+            storage_address BLOB
+        );
+        CREATE INDEX storage_addresses_storage_address ON storage_addresses (storage_address);
+        CREATE TABLE storage_updates_normalized (
+            block_number INTEGER REFERENCES canonical_blocks(number) ON DELETE CASCADE,
```

Comment:
@Mirko-von-Leipzig Do you think we still need to use `canonical_blocks` here? Why not simply `block_headers`?

Reply:
We can get rid of it now. We originally added it because we thought we wanted to keep fork block data as well, so canonical would store the pure blockchain without forks.

Reply:
I'd vote for re-creating that index as `UNIQUE`. From a correctness perspective `UNIQUE` is more precise anyway (and a _lot_ of our code would break if there were two headers with the same block number).

Reply:
Plus, if we ever want to really remove `canonical_blocks` the fewer references we have the better.

Reply:
Let's work towards only having canconical blocks removed and block headers primary key be the number 👍 This can be step one.